node/node.go

```diff
@@ -1152,6 +1153,16 @@ func (app *App) listenToUpdates(ctx context.Context) {
 				}
 				if len(update.Data.ActiveSet) > 0 {
 					app.hOracle.UpdateActiveSet(update.Data.Epoch, update.Data.ActiveSet)
+					app.eg.Go(func() error {
+						return atxsync.Download(
+							ctx,
+							10*time.Second,
+							app.addLogger(SyncLogger, app.log).Zap(),
+							app.db,
+							app.fetcher,
+							update.Data.ActiveSet,
+						)
+					})
```

Comment:
We are ranging over the `ch` channel which I don't think is correct, since so we can only return from this go routine directly after receiving an update (`<-ctx.Done()` is checked after receiving from `ch`). I fixed this already in this PR: https://github.com/spacemeshos/go-spacemesh/pull/5378 and the change for the incorrect ranging over the `ch` channel would incidentally also fix the linting issue here. 🙂

Reply:
The error returned from `atxsync.Download` should maybe also not be returned in the `eg.Go` function or no new go routines can be added to the group, but I believe the node will continue to run. Instead I think:

```suggestion
					app.eg.Go(func() error {
						if err := atxsync.Download(
							ctx,
							10*time.Second,
							app.addLogger(SyncLogger, app.log).Zap(),
							app.db,
							app.fetcher,
							update.Data.ActiveSet,
						); err != nil {
							app.errCh <- err
							return nil
						}
					})
```

might be correct if the node should shut down upon failing to get the ATXs from the active set.

Reply:
i fixed second comment, will leave 1st for your pr